orangecrab/src/WishboneUtils.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedRecordDot #-}
+{-# LANGUAGE NoFieldSelectors #-}
+
+module WishboneUtils where
+
+import Clash.Prelude
+import Protocols.Wishbone
+
+-- | Checks if the current `WishboneM2S` is in a wishbone cycle
+inWbCycle' ::
+  forall addrW selW dat.
+  -- | The wishbone instance
+  WishboneM2S addrW selW dat ->
+  -- | Wether or not we're in a wishbone cycle
+  Bool
+inWbCycle' wb = wb.strobe && wb.busCycle
+
+-- | Checks if the current `WishboneM2S` is in a wishbone cycle
+inWbCycle ::
+  forall dom addrW selW dat.
+  -- | The M2S signal
+  Signal dom (WishboneM2S addrW selW dat) ->
+  -- | Wether or not we're in a wishbone cycle
+  Signal dom Bool
+inWbCycle m2s = inWbCycle' <$> m2s
```

Comment:
I don't think this is a good pattern, it's barely shorter than the expressions they replace and these two are easy to mix up. I'd prefer if you remove these functions in favor of their respective `wb.strobe && wb.busCycle` and `wb.busCycle`